### ISSUE TITLE
Fix FreeBSD $OSTYPE check

### ISSUE THREAD

**Issue by ring-pete** [prowler-cloud/prowler#693]
I noticed that there was this issue previously opened:
https://github.com/toniblyx/prowler/issues/310

I believe there was a bug in that PR that was preventing it from being merged.  I believe this diff will correctly detect FreeBSD systems, and like macOS hosts use the appropriate gnu userland tools (gdate, gbase64 for example).  FreeBSD is similar to macOS in that it will also require the coreutils package to be installed (available as prebuild binary via "pkg"):

```
diff --git a/include/os_detector b/include/os_detector
index af962d4..023d47e 100644
--- a/include/os_detector
+++ b/include/os_detector
@@ -154,7 +154,7 @@ if [ "$OSTYPE" == "linux-gnu" ] || [ "$OSTYPE" == "linux-musl" ]; then
   test_tcp_connectivity() {
     gnu_test_tcp_connectivity "$1" "$2" "$3"
   }
-elif [[ "$OSTYPE" == "darwin"* ]]; then
+elif [[ "$OSTYPE" == "darwin"* ]] || [[ "$OSTYPE" == "freebsd"* ]]; then
   # BSD/OSX commands compatibility
   TEMP_REPORT_FILE=$(mktemp -t prowler.cred_report-XXXXXX)
   # It is possible that the user has installed GNU coreutils on OS X. By default, this will make GNU commands
```

Also note $OSTYPE evaluates to this on my 13-CURRENT FreeBSD system when using bash:
`freebsd13.0`

So it seems logical to use the wildcard to match just like with darwin.

**Comment by toniblyx** [comment]
Hi @ring-pete, thanks for the heads up with this. Can you please send a PR?

**Comment by ring-pete** [comment]
Thanks!  I've submitted this:
https://github.com/toniblyx/prowler/pull/698

**Comment by ring-pete** [comment]
Hello just bumping this issue to see if the PR looks good or not for committing.

**Comment by toniblyx** [comment]
Thanks, I´ll give it a look as soon as I can, sorry about that

**Comment by toniblyx** [comment]
This was fixed already in PR #698
